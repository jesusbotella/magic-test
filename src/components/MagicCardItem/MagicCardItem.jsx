import React from 'react';
import { StyleSheet, View, Text, TouchableHighlight, Image } from 'react-native';
import PropTypes from 'prop-types';
import ColorChip from '../ColorChip/ColorChip';

const styles = StyleSheet.create({
  card: {
    flexDirection: 'row',
    alignContent: 'center',
    padding: 16,
    borderBottomColor: '#e3e3e3',
    borderBottomWidth: 1
  },
  card__imgcontainer: {
    position: 'relative'
  },
  card__image: {
    width: 40,
    height: 55,
    marginRight: 10
  },
  card__name: {
    color: '#132B3B',
    fontSize: 20,
    fontWeight: '500',
    paddingBottom: 2
  },
  card__type: {
    color: '#164B7A',
    paddingTop: 2
  },
  card__color: {
    position: 'absolute',
    top: -5,
    right: 5
  }
});

export default function MagicCardItem(props) {
  const { card, onPress } = props;
  const { name, colors, type, setName, imageUrl } = card;

  return (
    <TouchableHighlight onPress={() => onPress(card)} underlayColor="#FAFAFA">
      <View style={styles.card}>
        <View style={{ flex: 1 }}>
          <Text style={styles.card__name}>{name}</Text>
          <Text>{type}</Text>
          <Text style={styles.card__type}>{setName}</Text>
        </View>

        <View style={styles.card__imgcontainer}>
          <Image source={{ uri: imageUrl }} style={styles.card__image} />

          <View style={styles.card__color}>
            {colors.map((color) => <ColorChip key={color} color={color} />)}
          </View>
        </View>
      </View>
    </TouchableHighlight>
  );
}

MagicCardItem.propTypes = {
  card: PropTypes.shape({
    name: PropTypes.string,
    colors: PropTypes.arrayOf(PropTypes.string),
    type: PropTypes.string,
    setName: PropTypes.string,
    imageUrl: PropTypes.string
  }),
  onPress: PropTypes.func
};

MagicCardItem.defaultProps = {
  card: {},
  onPress: () => {}
};
