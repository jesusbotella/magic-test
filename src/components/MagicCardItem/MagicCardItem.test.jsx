import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import MagicCardItem from './MagicCardItem';

const cardExample = {
  name: 'Abundance',
  colors: ['Green'],
  type: 'Enchantment',
  setName: 'Tenth Edition',
  imageUrl: 'http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=130483&type=card'
};

describe('MagicCardItem', () => {
  it('should render properly', () => {
    const element = render(<MagicCardItem card={cardExample} />).toJSON();
    expect(element).toMatchSnapshot();
  });

  it('should invoke onPress callback when pressed', () => {
    const onPressMock = jest.fn();

    const { getByTestId } = render(
      <MagicCardItem
        testID="pressTest"
        card={cardExample}
        onPress={onPressMock}
      />
    );

    fireEvent.press(getByTestId('pressTest'));

    // It should've been toHaveBeenCalledWith
    // but TouchableHighlight is hard to test
    expect(onPressMock).toHaveBeenCalled();
  });
});
