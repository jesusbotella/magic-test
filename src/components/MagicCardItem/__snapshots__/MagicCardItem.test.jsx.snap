// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MagicCardItem should render properly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "borderBottomColor": "#e3e3e3",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#132B3B",
            "fontSize": 20,
            "fontWeight": "500",
            "paddingBottom": 2,
          }
        }
      >
        Abundance
      </Text>
      <Text>
        Enchantment
      </Text>
      <Text
        style={
          Object {
            "color": "#164B7A",
            "paddingTop": 2,
          }
        }
      >
        Tenth Edition
      </Text>
    </View>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=130483&type=card",
          }
        }
        style={
          Object {
            "height": 55,
            "marginRight": 10,
            "width": 40,
          }
        }
      />
      <View
        style={
          Object {
            "position": "absolute",
            "right": 5,
            "top": -5,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "green",
              "borderColor": "#e3e3e3",
              "borderRadius": 50,
              "borderWidth": 2,
              "height": 10,
              "width": 10,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
